pkg/hashing/hashing.go

```diff
@@ -3,50 +3,24 @@ package hashing
 import (
 	"encoding/json"
 	"fmt"
+	utils "github.com/Yelp/paasta-tools-go/pkg/utils"
 	"hash/fnv"
 	"k8s.io/apimachinery/pkg/util/rand"
 	"reflect"
 )
 
 func ComputeHashForKubernetesObject(object interface{}) (string, error) {
-	// By marshaling/unmarshaling the object via JSON we're copying it into a
-	// map, which is easier to manipulate in generic way than structs.
-	if b, err := json.Marshal(object); err != nil {
-		return "", fmt.Errorf("Error while encoding %+v into JSON: %s", object, err)
+	if m, err := utils.GetHashObjectOfKubernetes(object); err != nil {
+		return "", err
 	} else {
-		var v map[string]interface{}
-		if err := json.Unmarshal(b, &v); err != nil {
-			return "", fmt.Errorf("Error while decoding JSON %s into an object: %s", v, err)
+		// By using serialized JSON for hashing we're making the hashing process
+		// a bit easier (like having maps always being sorted by keys).
+		if b, err := json.Marshal(m); err != nil {
+			return "", fmt.Errorf("Error while encoding %+v into JSON: %s", m, err)
```

Comment:
In the effort to leave the code nicer than when you found it, can you please make all the errors start with small letters rather than capital "Error" ? That's a Go style thing.

Reply:
Ack